tests/integration_tests/fixtures/source-test/source_test/run.py

```diff
@@ -65,6 +65,23 @@
                     },
                 },
             },
+            {
+                "name": "primary-key-with-dot",
+                "description": "This stream has a primary key with dot similar what is there in GAds.",
+                "source_defined_primary_key": [["table1.Column1"]],
+                "source_defined_cursor": False,
+                "supported_sync_modes": ["full_refresh"],
+                "json_schema": {
+                    "$schema": "http://json-schema.org/draft-07/schema#",
+                    "type": "object",
+                    "properties": {
+                        "table1.Column1": {"type": "string"},
+                        "table1.Column2": {"type": "number"},
+                        "table1.empty_column": {"type": "string"},
+                        "table1.big_number": {"type": "number"},
+                    },
```

Comment:
Thanks for adding this test! 🙏 

I think the schema we are trying to handle is slightly different...

```suggestion
                    "properties": {
                        "normal_field": {"type": "string"},
                        "nested": {
                            "type": "object",
                            "properties": {
                                "pk": {"type": "string"},
                                "other_val": {"type": "string"},
                            }
                            "required": "pk",
                        },
                    },
```

In this adaptation, the "." delimiter is a means of navigating to the subnode, but not actually part of the property names.

Reply:
My comment above is incorrect! Please disregard. I misunderstood the root cause. After digging into the source records, I found that the property name just contains the "." character, and so it is not "nested" in the way that I originally interpreted it - and as the guard statement in PyAirbyte was trying to prevent.

![image](https://github.com/user-attachments/assets/26f648ec-508f-4788-a206-8d9ec571bc07)

Reply:
Given my improved understanding now, it seems what is needed is just to escape the "." when we are normalizing the name, which I think would cause `ad_group.id` to become `ad_group_id`.

Reply:
I've opened a second PR targeting this one:

- https://github.com/pixisai/PyAirbyte/pull/9